packages/editor/src/plugins/drag-drop-bg/index.ts

```diff
@@ -0,0 +1,71 @@
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+
+import { DragDropBgEditor } from './editor'
+import {
+  type EditorPlugin,
+  type EditorPluginProps,
+  object,
+  string,
+  child,
+  number,
+  boolean,
+  list,
+} from '../../plugin'
+
+export const answerZoneImageProps = child({ plugin: EditorPluginType.Image })
+export const answerZoneTextProps = child({ plugin: EditorPluginType.Text })
+
+export const positionData = object({
+  top: number(0),
+  left: number(0),
+})
+
+export const layoutData = object({
+  width: number(0),
+  height: number(0),
+  visible: boolean(true),
+  lockedAspectRatio: boolean(true),
+})
+
+export const answerData = object({
+  image: answerZoneImageProps,
+  text: answerZoneTextProps,
+})
+
+export const answerZoneProps = object({
+  id: string(''),
+  position: positionData,
+  layout: layoutData,
+  answer: answerData,
+})
+
+export const wrongAnswerProps = object({
+  id: string(''),
+  answer: answerData,
+})
+
+const dragDropBgState = object({
+  answerZones: list(answerZoneProps, 0),
+  backgroundType: string(''),
+  backgroundImage: child({ plugin: EditorPluginType.Image }),
```

Comment:
Since there is a `backgroundType: 'text'` should this maybe be `optional()`?
And if there is only text/image as types we could maybe just remove `backgroundType` and use the image if it is defined.

Reply:
something like:
``` ts
const dragDropBgState = object({
  answerZones: list(answerZoneProps, 0),
  backgroundImage: optional(child({ plugin: EditorPluginType.Image })),
  extraDraggableAnswers: list(wrongAnswerProps, 0),
})
```

Reply:
I had difficulties dealing with optional state properties in other places, the defined() check or trying to get the value of the property seemed tricky.

We could maybe pair on this tomorrow if it is important

Reply:
> if it is important

Getting the state right is important imo (because it's quite annoying to change after there is content already created with an older version). But of course I don't know what the best state is :)
Let's see if we find a time for pairing.

In general it's a bit more code to work with optionals but it works quite well.
We have a lot of examples in the existing code.

Reply:
Made optional, thanks for the suggestion 👍